File: declined-node/src/resources/recoveries.ts
import type { HttpClient } from "../client.js";
import { API_PATHS, recoveryMarkRecoveredPath } from "../paths.js";
import type { ListParams } from "../types.js";

export class RecoveriesResource {
  constructor(private readonly http: HttpClient) {}

  list(params?: ListParams) {
    return this.http.request("GET", API_PATHS.recoveries, undefined, params);
  }

  markRecovered(recoveryAttemptId: string) {
    return this.http.request("POST", recoveryMarkRecoveredPath(recoveryAttemptId));
  }
}

File: declined-python/declined/resources/recoveries.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any

from declined.paths import API_PATHS, recovery_mark_recovered_path

if TYPE_CHECKING:
    from declined.client import HttpTransport


class RecoveriesResource:
    def __init__(self, http: HttpTransport):
        self._http = http

    def list(self, **params: Any) -> Any:
        return self._http.request("GET", API_PATHS["recoveries"], params=params or None)

    def mark_recovered(self, recovery_attempt_id: str) -> Any:
        return self._http.request(
            "POST",
            recovery_mark_recovered_path(recovery_attempt_id),
        )
